drivers/gpio/gpio_ite_it8xxx2_v2.c

```diff
@@ -162,6 +164,13 @@ static int gpio_ite_configure(const struct device *dev,
 	if (flags & GPIO_OUTPUT) {
 		ECREG(reg_gpcr) = (ECREG(reg_gpcr) | GPCR_PORT_PIN_MODE_OUTPUT) &
 				~GPCR_PORT_PIN_MODE_INPUT;
+		/*
+		 * gpioksi, gpioksoh and gpioksol require extended setting
+		 * when used as GPIO output.
+		 */
+		if (gpio_config->kbs_ctrl) {
+			ECREG(reg_gpcr) |= KSIX_KSOX_KBS_GPIO_MODE;
+		}
```

Comment:
This `KSIX_KSOX_KBS_GPIO_MODE` bit also needs to be set for GPIO inputs.  This is happening by accident because `GPCR_PORT_PIN_MODE_INPUT` and `KSIX_KSOX_KBS_GPIO_MODE` are both bit 7.

When `kbs_ctrl` is true, you're actually writing to the KSxxGCTRLR register and not a GPCR register.
The code should make this clear without relying on bit masks being the same:
```
if (kbs_ctrl) {
    ksxgctrlr = ECREG(reg_gpcr);
    ksxgctrlr |= KSIX_KSOX_KBS_GPIO_MODE;
    if (OUTPUT) {
        ksxgctrlr |= KSIX_KSOX_GPIO_OUTPUT;
    } else {
        ksxgctrlr &= ~KSIX_KSOX_GPIO_OUTPUT;
    }
    ECREG(reg_gpcr) = ksxgtrlr;
} else {
/* Handle regular GPIO controller writes to GPCR */    
```

For completeness, you should also include the pull-up/pull-down configuration in the same `if (kbs_ctrl)` block.

Reply:
Done.